### PR TITLE
SignIn and Auth

### PR DESCRIPTION
**Pull request by andresgpr**
This PR will allow sign-in and auth routing.

### CODE REVIEW

File: server/controllers/auth.controller.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import User from '../models/user.model.js'
+import jwt from 'jsonwebtoken'
```
Comment (Curry091104) [inline server/controllers/auth.controller.js:2]: This made app crashed. Please check again

node:internal/errors:490
[1]     ErrorCaptureStackTrace(err);
[1]     ^
[1] 
[1] Error [ERR_MODULE_NOT_FOUND]: Cannot find package 'jsonwebtoken' imported from D:\Centennial College Materials\THIRD SEMESTER\COMP 229 - WEB APPLICATION DEVELOPMENT\GROUP PROJECT\The Tomorrow Team - The Brainy Book Store\COMP229_GroupProject\server\controllers\auth.controller.js

**Review by Curry091104 — COMMENTED** [review]
App crashed

**Comment by Curry091104** [comment]
{
    "error": "Could not sign in"
}
I think we need to add something to the user.model.js

File: server/controllers/auth.controller.js (hunk 1 above)
Comment (Curry091104) [reply server/controllers/auth.controller.js:2]: I solved this issue. We missed jsonwebtoken dependency in package.json

**Comment by wowankit378** [comment]
i couldnot sign in

**Review by wowankit378 — COMMENTED** [review]
i think its good

**Comment by Curry091104** [comment]
> i think its good

Can you show me the way you tested to sign in?

**Comment by andresgpr** [comment]
> {
> "error": "Could not sign in"
> }

Do you install express-jwt and jsonwebtoken? can you share an screenshot of how you're testing when getting this error?

@Curry091104

**Comment by Curry091104** [comment]
I solved the problem. I updated in user.model.js

**Comment by andresgpr** [comment]
> I solved the problem. I updated in user.model.js. We hadn't had important codes for password validation

Yes, It's missing the authenticate method.

**Comment by Curry091104** [comment]
I think we can merge them together

**Pull request merged**